### PR TITLE
Feat/playlist sync

### PR DESCRIPTION
**Pull request by aricalves**
# This is a big one, expect alot of merge conflicts.
#### This PR contains:
 - Introduce the `react-youtube` module, which helps us create a custom YouTube player
 - Refactor `VideoPlayer` component to handle database formatted video data
 - Implement synchronized playlist mechanics

### [Trello Ticket](https://trello.com/c/pk4fjgdC)

### CODE REVIEW

File: client/src/components/Room/RoomView.jsx [hunk 1]
```diff
@@ -59,18 +71,29 @@ class RoomView extends React.Component {
 
   renderPlaylist() {
     return axios.get('/renderPlaylist')
-      .then(response => this.setState({ playlist: response.data }))
+      .then(response => this.setState({
+        playlist: response.data,
+        currentVideo: response.data[0],
```
Comment (sorbae) [inline client/src/components/Room/RoomView.jsx]: Will need to eventually change currentVideo to the video at playlist's current index, which will be stored and incremented in the database.

File: client/src/components/Room/RoomView.jsx [hunk 2]
```diff
@@ -59,18 +71,29 @@ class RoomView extends React.Component {
...
+      }))
       .catch(err => console.error('Could not retreive playlist: ', err));
   }
 
   render() {
     return (
       <div className="room">
-        <div className="container navbar">fam.ly</div>
+        <div className="container navbar">__DEVELOPMENT</div>
```
Comment (sorbae) [inline client/src/components/Room/RoomView.jsx]: Aric, please.

File: client/src/components/Room/RoomView.jsx [hunk 3]
```diff
@@ -59,18 +71,29 @@ class RoomView extends React.Component {
...
         <Playlist playlist={this.state.playlist} />
-        <VideoPlayer currentVideo={this.state.currentVideo} />
+        <VideoPlayer
+          loadPlayer={this.loadPlayer}
```
Comment (sorbae) [inline client/src/components/Room/RoomView.jsx]: Where is loadPlayer being defined?

File: client/src/components/Room/VideoPlayer.jsx [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
-      title="videoPlaying"
-      className="videoFrame"
-      frameBorder="0"
-      src={`https://www.youtube.com/embed/${currentVideo.id.videoId}?autoplay=0`} // TODO turn autoplay back on for deployment
-      allowFullScreen
-    />
-    <div className="currentSongDetails">
-      <div className="videoTitle"> {currentVideo.snippet.title} </div>
-      <div className="songDescription"> {currentVideo.snippet.description} </div>
+const VideoPlayer = props =>
```
Comment (sorbae) [inline client/src/components/Room/VideoPlayer.jsx]: We should probably stick to the style of deconstructioning of props?

File: client/src/components/Room/VideoPlayer.jsx [hunk 5]
```diff
@@ -1,26 +1,28 @@
...
+  (
```
Comment (sorbae) [inline client/src/components/Room/VideoPlayer.jsx]: Move opening parenthesis back up to the previous line to match our coding style thus far.

File: client/src/components/Room/RoomView.jsx (hunk 1 above)
Comment (aricalves) [reply client/src/components/Room/RoomView.jsx]: you are correct

File: client/src/components/Room/RoomView.jsx (hunk 2 above)
Comment (aricalves) [reply client/src/components/Room/RoomView.jsx]: Woops

File: client/src/components/Room/RoomView.jsx (hunk 3 above)
Comment (aricalves) [reply client/src/components/Room/RoomView.jsx]: woops

File: client/src/components/Room/VideoPlayer.jsx (hunk 4 above)
Comment (aricalves) [reply client/src/components/Room/VideoPlayer.jsx]: correct

File: client/src/components/Room/VideoPlayer.jsx (hunk 5 above)
Comment (aricalves) [reply client/src/components/Room/VideoPlayer.jsx]: k

**Pull request merged**